### PR TITLE
[virt-launcher]: Issue an error for disk with DedicatedIOThread and SATA bus

### PR DESCRIPTION
**Pull request by iholder-redhat**
**What this PR does / why we need it**:
Livirt does not support disks with SATA and DedicatedIOThread. Currently VMs with such configurations are failing to boot with Libvirt error.

This PR will lead to the the user seeing Kubevirt error instead of Libvirt error.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
PR Addresses the following Bugzilla ticket: https://bugzilla.redhat.com/show_bug.cgi?id=1911590

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go [hunk 1]
```diff
@@ -1764,6 +1764,18 @@ func validateDisks(field *k8sfield.Path, disks []v1.Disk) []metav1.StatusCause {
...
 			}
+
+			// Reject defining DedicatedIOThread to a disk with SATA bus since this configuration
+			// is not supported in libvirt.
+			isIOThreadsWithSataBus := disk.DedicatedIOThread != nil && *disk.DedicatedIOThread &&
+				(disk.DiskDevice.Disk != nil) && strings.EqualFold(disk.DiskDevice.Disk.Bus, "sata")
+			if isIOThreadsWithSataBus {
+				causes = append(causes, metav1.StatusCause{
+					Type:    metav1.CauseTypeFieldValueNotSupported,
+					Message: fmt.Sprintf("Disk with DedicatedIOThread and a SATA bus is not supported"),
```
Comment (vladikr) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go:1775]: How would you feel about changing the error message to something like "IOThreads are not supported for disks on a SATA bus." ?

**Review by vladikr — COMMENTED** [review]
Thanks for the fix!
Everything makes sense. Just a minor suggestion to change the error message.
/approve

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go (hunk 1 above)
Comment (iholder-redhat) [reply pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go:1775]: Indeed your message is clearer. I'll change that. Thanks! :)

**Comment by vladikr** [comment]
/lgtm cancel
@iholder-redhat please look at the unit tests

**Review by dhiller — APPROVED** [review]
/lgtm

**Pull request merged**